spec/models/ext_management_system_spec.rb

```diff
@@ -404,71 +404,65 @@
 
-    context "with no child managers" do
-      let(:ems) do
-        FactoryGirl.create(:ext_management_system, :zone => zone)
-      end
-
-      it "returns a task" do
-        task_id = ems.destroy_queue
+    it "queues up destroy" do
+      described_class.destroy_queue([ems.id, ems2.id])
 
-        deliver_queue_message
-
-        task = MiqTask.find(task_id)
-        expect(task.state).to  eq("Finished")
-        expect(task.status).to eq("Ok")
-      end
-
-      it "re-schedules with active workers" do
-        FactoryGirl.create(:miq_ems_refresh_worker, :queue_name => ems.queue_name, :status => "started", :miq_server => server)
-        ems.destroy_queue
-
-        expect(MiqQueue.count).to eq(1)
+      expect(MiqQueue.count).to eq(2)
+      expect(MiqQueue.pluck(:instance_id)).to include(ems.id, ems2.id)
+    end
+  end
 
-        deliver_queue_message
+  context "#destroy_queue" do
+    let(:server) { EvmSpecHelper.local_miq_server }
+    let(:zone)   { server.zone }
+    let(:ems)    { FactoryGirl.create(:ext_management_system, :zone => zone) }
 
-        expect(MiqQueue.count).to eq(1)
-        expect(MiqQueue.last.deliver_on).to_not be_nil
-      end
+    it "returns a task" do
+      task_id = ems.destroy_queue
 
-      it "destroys the ems when the active worker shuts down" do
-        refresh_worker = FactoryGirl.create(:miq_ems_refresh_worker, :queue_name => ems.queue_name, :status => "started", :miq_server => server)
-        ems.destroy_queue
+      deliver_queue_message
 
-        deliver_queue_message
+      task = MiqTask.find(task_id)
+      expect(task.state).to  eq("Finished")
+      expect(task.status).to eq("Ok")
+    end
 
-        expect(ExtManagementSystem.count).to eq(1)
+    it "destroys the ems when no active worker exists" do
+      ems.destroy_queue
 
-        refresh_worker.destroy
+      expect(MiqQueue.count).to eq(1)
 
-        deliver_queue_message
+      deliver_queue_message
 
-        expect(ExtManagementSystem.count).to eq(0)
-      end
+      expect(MiqQueue.count).to eq(0)
+      expect(ExtManagementSystem.count).to eq(0)
     end
 
-    context "with child managers" do
-      let(:child_manager) { FactoryGirl.create(:ext_management_system) }
-      let(:ems)           { FactoryGirl.create(:ext_management_system, :zone => zone, :child_managers => [child_manager]) }
+    it "destroys the ems when active worker exists" do
+      FactoryGirl.create(:miq_ems_refresh_worker, :queue_name => ems.queue_name, :status => "started", :miq_server => server)
+      ems.destroy_queue
 
-      it "queues up destroy for child_managers" do
-        described_class.destroy_queue(ems.id)
+      expect(MiqQueue.count).to eq(1)
 
-        expect(MiqQueue.count).to eq(2)
-        expect(MiqQueue.pluck(:instance_id)).to include(ems.id, child_manager.id)
-      end
+      deliver_queue_message
+
+      expect(MiqQueue.count).to eq(0)
+      expect(ExtManagementSystem.count).to eq(0)
```

Comment:
Do you also need an expectation that the worker is gone?

Reply:
👍